students/JackieCao/Lesson04/trigrams.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python3
+"""
+Trigrams
+"""
+import random
+import sys
+
+def read_book(filename):
+    with open(filename) as f:
+        lines = [line.rstrip('\n') for line in f]
+    return lines
+
+def make_words(book, stat_line, end_line):
+    # specify the start and ending line (exclusive)
+    n = [i for i,x in enumerate(book) if x == start_line] 
```

Comment:
this is looping through the book twice, and not really a list comp, as you don't actually want a list when you're done.

clever though.